File: Data Structures & Algorithms/is-palindrome/submission-3.js
class Solution {
    isPalindrome(s) {
        let l = 0, r = s.length - 1

        while (l < r) {
            while (l < r && !this.isAlphaNum(s[l])) l++
            while (l < r && !this.isAlphaNum(s[r])) r--

            if (s[l].toLowerCase() !== s[r].toLowerCase()) {
                return false
            }

            l++
            r--
        }

        return true
    }

    isAlphaNum(c) {
        return c.toLowerCase() !== c.toUpperCase() || (c >= '0' && c <= '9')
    }
}

File: Data Structures & Algorithms/palindrome-number/submission-3.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        strNum = str(x)

        l = 0
        r = len(strNum) - 1

        while l < r:
            if strNum[l] != strNum[r]:
                return False

            l += 1
            r -= 1

        return True